lib/msal-angular/src/msal-guard.service.ts

```diff
@@ -38,30 +83,14 @@ export class MsalGuard implements CanActivate {
         }
 
         if (!this.authService.getAccount()) {
-            if (this.msalAngularConfig.popUp) {
-                return this.authService.loginPopup({
-                    scopes: this.msalAngularConfig.consentScopes,
-                    extraQueryParameters: this.msalAngularConfig.extraQueryParameters
-                })
-                    .then(() => true)
-                    .catch(() => false);
-            }
-
-            const routePath = `${window.location.origin}${state.url}`;
-
-            this.authService.loginRedirect({
-                redirectStartPage: routePath,
-                scopes: this.msalAngularConfig.consentScopes,
-                extraQueryParameters: this.msalAngularConfig.extraQueryParameters
-            });
-        } else {
-            return this.authService.acquireTokenSilent({
-                scopes: [this.msalConfig.auth.clientId]
-            })
-                .then(() => true)
-                .catch(() => false);
+            return this.loginInteractively(state.url);
         }
 
+        return this.authService.acquireTokenSilent({
+            scopes: [this.msalConfig.auth.clientId]
+        })
+            .then(() => true)
+            .catch(() => this.loginInteractively(state.url));
```

Comment:
This probably should check if the error requires interaction, and only invoke `loginInteractively` if so.

Reply:
+1

Reply:
Done.